src/poetry/console/commands/build.py

```diff
@@ -47,49 +59,120 @@ class BuildCommand(EnvCommand):
         "poetry.core.masonry.builders.wheel",
     ]
 
-    def _build(
-        self,
-        fmt: str,
-        executable: str | Path | None = None,
-        *,
-        target_dir: Path | None = None,
-    ) -> None:
-        from poetry.masonry.builders import BUILD_FORMATS
+    def _requested_formats(self) -> list[Literal["sdist", "wheel"]]:
+        fmt = self.option("format") or "all"
 
         if fmt in BUILD_FORMATS:
-            builders = [BUILD_FORMATS[fmt]]
+            formats = [fmt]
         elif fmt == "all":
-            builders = list(BUILD_FORMATS.values())
+            formats = BUILD_FORMATS
         else:
             raise ValueError(f"Invalid format: {fmt}")
 
+        return formats  # type: ignore[return-value]  # TODO
+
+    def _config_settings(self) -> dict[str, str]:
+        config_settings = {}
+
         if local_version_label := self.option("local-version"):
-            self.poetry.package.version = self.poetry.package.version.replace(
-                local=local_version_label
+            self.line_error(
+                f"<warning>`<fg=yellow;options=bold>--local-version</>` is deprecated."
```

Comment:
There might be some push back to this.

Reply:
I wasn't sure here. We can keep it and document that this is a shorthand for `--config_settings local-version=...`. But people should be aware that `-local-version` only works if the desired backend have a config_settings with that name.

Reply:
Is there any "standardisation" on the config settings, like they key for "local-version"? 

But yeah we could mark it for deprecation and reverse if there is a lot of push back, consider how we want to deal with it. 

Alternatively, if would be good if we can detect if they are using poetry masonry api, and if so skip the log.

Reply:
> Is there any "standardisation" on the config settings, like they key for "local-version"?

Not that I know. This is something that can be decided by the build backends.